jiant/utils/retokenize.py

```diff
@@ -179,31 +181,91 @@ def project_tokens(self, idxs: Union[int, Sequence[int]]) -> Sequence[int]:
         """
         if isinstance(idxs, int):
             idxs = [idxs]
-        return self.T[idxs].nonzero()[1]  # column indices
-
-    def project_span(self, start, end) -> Tuple[int, int]:
+        return self.source_token_idx_to_target_token_idx[idxs].nonzero()[1]  # column indices
+
+    @classmethod
+    def _project_span(cls, mat, start, end, inclusive):
```

Comment:
Please add a test for this — here's a test I wrote to check the logic (it assumes you make `_project_span` static):

```
def test_private_project_token_span():
    mat = np.eye(5, dtype=int)
    mat[0][0] = 0
    mat[3][3] = 0
    assert TokenAligner._project_span(mat, 1, 3, inclusive=True) == (1, 2)
    assert TokenAligner._project_span(mat, 1, 3, inclusive=False) == (1, 3)
    assert TokenAligner._project_span(mat, 1, 2, inclusive=True) == (1, 2)
    assert TokenAligner._project_span(mat, 1, 2, inclusive=False) == (1, 2)
    assert TokenAligner._project_span(mat, 1, 4, inclusive=True) == (1, 4)
    assert TokenAligner._project_span(mat, 1, 4, inclusive=False) == (1, 3)

```

Reply:
thanks!